src/planning.py

```diff
@@ -224,16 +238,23 @@ def _skeleton_generator(
                             heuristic(child_node.atoms))
                 hq.heappush(queue, (priority, rng_prio.uniform(), child_node))
     if not queue:
-        raise ApproachFailure("Planning ran out of skeletons!")
+        raise _MaxSkeletonsFailure("Planning ran out of skeletons!")
     assert time.time() - start_time >= timeout
-    raise ApproachTimeout("Planning timed out in skeleton search!")
+    raise _SkeletonSearchTimeout
 
 
-def _run_low_level_search(task: Task, option_model: _OptionModelBase,
-                          skeleton: List[_GroundNSRT],
-                          atoms_sequence: List[Set[GroundAtom]], seed: int,
-                          timeout: float) -> Optional[List[_Option]]:
-    """Backtracking search over continuous values."""
+def _run_low_level_search(
+        task: Task, option_model: _OptionModelBase,
+        skeleton: List[_GroundNSRT], atoms_sequence: List[Set[GroundAtom]],
+        seed: int,
+        timeout: float) -> Tuple[Optional[List[_Option]], List[_Option]]:
+    """Backtracking search over continuous values.
```

Comment:
Wouldn't it make more sense to collapse these return values into one, and just have the caller check whether the length of the returned plan equals the length of the skeleton? (If so, that's a success.)

Reply:
done